### ISSUE TITLE
paket not extracting package with local override

### ISSUE THREAD

**Issue by daz10000** [fsprojects/Paket#2839]
### Description

Using paket.local to pull in a package from a local build,  the nupkg file is copied but not extracted

### Repro steps

Messy to reproduce without the entire project, but 

1. Built upstream project with project scaffold ./build.sh BuildPackage

2. add paket.local file to dependent project

`nuget GslCore group Build -> source c:\extproj\GslCore\bin`

Run .paket/paket.exe restore on dependent project

Build dependent project (and have it work)

### Expected behavior
I would expect paket to copy over the GslCore.nupkg file and unpack it.  

### Actual behavior

When the dependent project is built, it is referring to packages/build/GslCore/lib/net461/GslCore.dll    and those references all fail because the lib and lower folders are missing as paket is not extracting the nupkg.    If I remove the whole packages folder and children, then do `paket.exe restore --verbose --force` and look at the output, the only steps for this library reported are

```
$ grep GslCore out
paket.local override: nuget GslCore group Build -> c:\extproj\GslCore\bin
Cleaning C:\src\dem_gslc\packages\build\GslCore
GslCore 0.4.28 already extracted
```

It looks like it is (wrongly?) concluding that the GslCore package is already extracted and failing to extract it.

### Known workarounds

I'm stuck - I won't be able to use paket.local unless I can fix this.

**Comment by daz10000** [comment]
So I worked out why this is happening by instrumenting the code that checked isExtracted.   It's looking for files that aren't the original package or the license file, and in my case the original package file is capitalized (GslCore.0.4.28.nupkg) and the file it's excluding is gslcore.0.4.28.nupkg so it sees the package file and incorrectly thinks it is extracted.  I will check all my naming, but everything should be referred to by the camelcaps form, so I think paket must be forcing lowercase somewhere, and the fix would be to lowercase the files as well as they are checked.  I'll try that now as a workaround, but is that a safe fix?

Darren


Debug: directory=C:\src\dem_gslc\packages\build\GslCore licenseFile=C:\src\dem_gslc\packages\build\GslCore\gslcore.0.4.28.license.html  // ignores license file
 fullName=C:\src\dem_gslc\packages\build\GslCore\gslcore.0.4.28.nupkg  // ignores this file
Debug:  f=C:\src\dem_gslc\packages\build\GslCore\GslCore.0.4.28.nupkg  // finds this file

**Comment by daz10000** [comment]
I can confirm that fix works.  I just added .ToLower() to the checks at the end.  This still isn't quite unblocking my project because paket is so good about downloading and updating itself.   Is there any easy way of disabling that check in a projectscaffold build script?  I tested the recompiler paket standalone but can't get my project to build still b/c paket updates

Darren
p.s. trying to put together a pull request but stuck in a court waiting jury duty with 7kb/second wifi.  Not a great day.

```let inline isExtracted (directory:DirectoryInfo) (packageName:PackageName) (version:SemVerInfo) =
    let inDir f = Path.Combine(directory.FullName, f)
    let packFile = GetPackageFileName packageName version |> inDir
    let licenseFile = GetLicenseFileName packageName version |> inDir
    let fi = FileInfo(packFile)
    if not fi.Exists then false else
    if not directory.Exists then false else
    directory.EnumerateFileSystemInfos()
    |> Seq.exists (f.FullName.ToLower() <> fi.FullName.ToLower() && f.FullName.ToLower() <> licenseFile.ToLower())
```

**Comment by forki** [comment]
Can you please send a pull request?
If you don't want paket to override itself the make sure you don't call the
paket bootstrapper.exe

Am 12.10.2017 18:27 schrieb "daz10000" <notifications@github.com>:

I can confirm that fix works. I just added .ToLower() to the checks at the
end. This still isn't quite unblocking my project because paket is so good
about downloading and updating itself. Is there any easy way of disabling
that check in a projectscaffold build script? I tested the recompiler paket
standalone but can't get my project to build still b/c paket updates

Darren
p.s. trying to put together a pull request but stuck in a court waiting
jury duty with 7kb/second wifi. Not a great day.

    let inDir f = Path.Combine(directory.FullName, f)
    let packFile = GetPackageFileName packageName version |> inDir
    let licenseFile = GetLicenseFileName packageName version |> inDir
    let fi = FileInfo(packFile)
    if not fi.Exists then false else
    if not directory.Exists then false else
    directory.EnumerateFileSystemInfos()
    |> Seq.exists (f.FullName.ToLower() <> fi.FullName.ToLower() &&
f.FullName.ToLower() <> licenseFile.ToLower())

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/fsprojects/Paket/issues/2839#issuecomment-336191207>,
or mute the thread

**Comment by daz10000** [comment]
See PR #2842  - thanks!

**Comment by enricosada** [comment]
Fix by PR #2842 , thanks @daz10000 !

Closing
